docs/generate-metadata.js

```diff
@@ -0,0 +1,98 @@
+import metadata from 'react-component-metadata';
+import glob from 'glob';
+import fsp from 'fs-promise';
+
+let globp = promisify(glob);
+
+let cleanDoclets = desc => {
+  let idx = desc.indexOf('@');
+  return (idx === -1 ? desc : desc.substr(0, idx )).trim();
+};
+
+let cleanDocletValue = str => str.replace(/^\{/, '').replace(/\}$/, '');
+
+let isLiteral = str => str[0] === '"' || str[0] === "'";
+
+// override parsed metadata with jsDoc doclets
+function applyDoclets(props, propName){
+  let prop = props[propName];
+  let doclets = prop.doclets;
+  let value;
+
+  if (doclets.type) {
+    value = cleanDocletValue(doclets.type);
+    prop.type.name = value;
+
+    if ( value[0] === '(' ) {
+      value = value.substring(1, value.length - 1).split('|');
+
+      prop.type.value = value;
+      prop.type.name = value.every(isLiteral) ? 'enum' : 'union';
+    }
+  }
+
+  if ( doclets.required) {
+    prop.required = true;
+  }
+}
+
+export default function generate(destination, options = { mixins: true }){
+  //console.log(path.join(__dirname, '../src/**/*.js'))
+
+  return globp(__dirname + '/../src/**/*.js') //eslint-disable-line no-path-concat
+    .then( files => {
+      //console.log(files)
+      let results = files.map(
+        filename => fsp.readFile(filename).then(content => metadata(content, options)) );
+
+      return Promise.all(results)
+        .then( data => {
+          let result = {};
+
+          data.forEach(components => {
+            Object.keys(components).forEach(key => {
+              const component = components[key];
+              component.doclets = metadata.parseDoclets(component.desc || '');
+              component.desc = cleanDoclets(component.desc || '');
+
+              Object.keys(component.props).forEach( propName => {
+                const prop = component.props[propName];
+
+                prop.doclets = metadata.parseDoclets(prop.desc || '');
+                prop.desc = cleanDoclets(prop.desc || '');
+
+                applyDoclets(component.props, propName);
+              });
+            });
+
+            result = { ...result, ...components };
+          });
+
+          //console.log(result);
+
+          return result;
+        })
+        .catch( e => setTimeout(()=> { throw e; }));
+    });
+}
+
+
+function promisify(fn){
+  return function (...args){
+    return new Promise(function(resolve, reject){
+
+      function finish(err, result){
+        //console.log('hi', result)
+        if (err) {
+          return reject(err);
+        }
+        resolve(result);
+      }
+
+      args = args.concat(finish);
+
+      fn.apply(null, args);
+    });
+
+  };
+}
```

Comment:
Should this be defined somewhere else? I think it's simple enough that we shouldn't need to find another dependency that provides it, especially since this is for dev tools only. Though it is generic enough to move to say `tools/promisify` maybe?